Reset application.db and  wasm

chillyvee opened the pull request:
Is there a better way to reset application.db and wasm?  For example by extending into the "tendermint" reset-state command?

the-frey commented:
This looks potentially useful.

Would also say that it's worth maybe removing cache as well in a separate command 🤔 

There have been a bunch of changes on main though so might be easier to start fresh and just add these changes (or rebase)

chillyvee commented:
Can take another look soon since testnet completed the upgrade.

chillyvee commented:
> Would also say that it's worth maybe removing cache as well in a separate command 🤔

Split the `application.db` and `wasm` reset commands

Rebased on main

vuong177 commented on cmd/junod/tendermint.go:
```diff
@@ -0,0 +1,79 @@
+		if err := os.RemoveAll(wasmDir); err == nil {
+			fmt.Println("Removed wasm", "dir", wasmDir)
+		} else {
+			fmt.Println("error removing wasm", "dir", wasmDir, "err", err)
```
return fmt.Errorf("error removing wasm %s dir: err %w", wasmDir, err)

vuong177 replied:
```suggestion
			return fmt.Errorf("error removing wasm  dir: %s; err: %w", wasmDir, err)
```

vuong177 commented on cmd/junod/tendermint.go:
```diff
@@ -0,0 +1,79 @@
+		if err := os.RemoveAll(appDir); err == nil {
+			fmt.Println("Removed application.db", "dir", appDir)
+		} else {
+			fmt.Println("error removing application.db", "dir", appDir, "err", err)
```
```suggestion
			return fmt.Errorf("error removing application.db  dir: %s; err: %w", wasmDir, err)

vuong177 commented on cmd/junod/tendermint.go:
```diff
@@ -0,0 +1,79 @@
+	}
+
+	if err := tmos.EnsureDir(appDir, 0700); err != nil {
+		fmt.Println("unable to recreate application.db", "err", err)
```
same nits
```suggestion
		return fmt.Errorf("unable to recreate application.db %w", err)
```

vuong177 commented on cmd/junod/tendermint.go:
```diff
@@ -0,0 +1,79 @@
+	}
+
+	if err := tmos.EnsureDir(wasmDir, 0700); err != nil {
+		fmt.Println("unable to recreate wasm", "err", err)
```
same nits
```suggestion
		return fmt.Errorf("unable to recreate wasm %w", err)
```

vuong177 commented:
@chillyvee, thanks for making this. Only some minors left, I will approve when it is addressed.

chillyvee commented:
@vuong177 Thank you for the feedback.  We'll get this amended as soon as we have a chance.

chillyvee commented:
@vuong177 Your suggestions have been added to the branch

Reecepbcups reviewed (approved):
Approved but my only thought is should we have this in the root commands?
or make its own reset section? `junod reset app`, `junod reset wasm` @chillyvee 

In the future we may want to add resetting config files like app.toml as well to default
